lib/stat_tracker.rb

```diff
@@ -31,25 +31,46 @@ def self.from_csv(locations)
 
     stat_tracker = StatTracker.new(games, teams, game_teams)
   end
-  # Game Statistics
-  def highest_total_score
-    high_score = 0
-    @games.each_value do |game|
-      high = game.away_goals + game.home_goals
-      if (game.away_goals + game.home_goals) > high_score
-        high_score = game.away_goals + game.home_goals
-      end
-    end
-    high_score
+
+  def count_of_teams
+    @teams.length
   end
-  # League Statistics
 
+  def best_offense
+   id = @game_teams.max_by do |team|
+          average_goals_per_game(team.team_id)
+        end.team_id
 
-  # Season Statistics
+    @teams.find do |team|
+      id == team.team_id
+    end.team_name
+  end
 
+  def games_by_team(team_id)
+    @game_teams.select do |game|
+      game.team_id == team_id
+    end
+  end
 
-  # Team Statistics
+  def total_goals_by_team(team_id)
+    goals = []
+    games_by_team(team_id).each do |game|
+      goals << game.goals
+    end
+    goals.sum
+  end
 
+  def average_goals_per_game(team_id)
+    total_goals_by_team(team_id).to_f / games_by_team(team_id).length.to_f
+  end
 
+  def worst_offense
+    id = id = @game_teams.min_by do |team|
```

Comment:
hmmmm....you assign the ID variable twice here.  I don't think it hurts your code or changes anything, but we should probably clean that up anyways. :)

Reply:
Oooop, good catch! That's what I get for trying to write code after midnight haha. Eagle eyes at it again, thank you!!